contracts/Market.sol

```diff
@@ -0,0 +1,94 @@
+pragma solidity ^0.4.24;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
+
+import "./IMarket.sol";
+import "./Property.sol";
+
+// @title Radical Bodies market.
+contract Market is IMarket, Ownable, Pausable {
+    Property private _token;
+    uint256 private _taxRate;
+    uint256 private _epsilon;
+    address private _beneficiary;
+
+    constructor(
+        Property token,
+        uint256 taxRate,
+        uint256 epsilon,
+        address beneficiary
+    ) {
+        _token = token;
+        _taxRate = taxRate;
+        _epsilon = epsilon;
+        _beneficiary = beneficiary;
+    }
+
+    // The ERC721 token.
+    function token() external view returns (address) {
+        return address(_token);
+    }
+
+    // The tax rate in basis points, 0 to 1000000.
+    function taxRate() external view returns (uint256) {
+        return _taxRate;
+    }
+
+    // The minimum bid/increase. (in ETH)
+    function epsilon() external view returns (uint256) {
+        return _epsilon;
+    }
+
+    // The beneficiary receiving all the funds.
+    function beneficiary() external view returns (address) {
+        return _beneficiary;
+    }
+
+    // The beneficiary can retrieve already released funds via this.
+    function withdrawAvailableFunds() external payable {
```

Comment:
We could do an on behalfOf function here, ```withdrawAvailableFundsFor```, would allow for us to provide the gas.

Reply:
Let's change that in a separate PR then, changing `IMarket` at the same time.